src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -27,4 +27,30 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public int hashCode() {
+        int result = 347;
+        result = result * 743 + ((width == null) ? 0 : width.hashCode());
+        result = result * 743 + ((length == null) ? 0 : length.hashCode());
+        result = result * 743 + ((color == null) ? 0 : color.hashCode());
+        return result;
+    }
+
+    @Override
+    public boolean equals(Object object) {
+        if (this == object) {
+            return true;
+        }
+        if (object == null) {
+            return false;
+        }
+        if (object.getClass().equals(Rectangle.class)) {
+            Rectangle rectangle = (Rectangle) object;
+            return width == null || width.equals(rectangle.width)
+                    && length == null || length.equals(rectangle.length)
+                    && color == null || color.equals(rectangle.color);
```

Comment:
what if width == null and rectangle.width != null?

Reply:
Fixed. Trying to remove Main from push.